protocol/app/middleware/middleware_test.go

```diff
@@ -1,68 +1,73 @@
 package middleware_test
 
 import (
+	"bytes"
 	"fmt"
+	"github.com/cometbft/cometbft/libs/log"
 	"testing"
 
 	"github.com/dydxprotocol/v4-chain/protocol/app/middleware"
-	"github.com/dydxprotocol/v4-chain/protocol/mocks"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 )
 
 func TestRunTxPanicLoggingMiddleware(t *testing.T) {
-	logger := &mocks.Logger{}
-	middleware.Logger = logger
+	tests := map[string]struct {
+		function     func()
+		expectedLogs []string
+	}{
+		"no panic": {
+			function: func() {
+				// Do something that does not panic.
+			},
+		},
+		"panic with string": {
+			function: func() {
+				panic("test123")
+			},
+			expectedLogs: []string{
+				"E[202",                       // error and date prefix
+				"runTx panic'ed with test123", // message
+				"middleware_test.go",          // part of stack trace
+			},
+		},
+		"panic with error": {
+			function: func() {
+				panic(fmt.Errorf("test456"))
+			},
+			expectedLogs: []string{
+				"E[202",                       // error and date prefix
+				"runTx panic'ed with test456", // message
+				"middleware_test.go",          // part of stack trace
+			},
+		},
+	}
 
-	func() {
-		defer func() {
-			if r := recover(); r != nil {
-				handler := middleware.NewRunTxPanicLoggingMiddleware()
-				err := handler(r)
-				require.Nil(t, err)
-			}
-		}()
-		// Do something that does not panic.
-	}()
-	logger.AssertExpectations(t)
-
-	logger.On(
-		"Error",
-		"runTx panic'ed with a problem",
-		"stack trace",
-		mock.Anything,
-	).Return(nil).Once()
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Restore the old logger after the test runs
+			oldLogger := middleware.Logger
+			defer func() { middleware.Logger = oldLogger }()
```

Comment:
for my understanding, why do we need to restore the logger? is it to reset the logger for each test run?

Reply:
middleware.Logger is a global variable, clarified in comment